x/oracle/keeper/oracle_test.go

```diff
@@ -206,17 +225,118 @@ func (suite *oracleTestSuite) TestRegisterOracleNotSameUniqueID() {
+	}
+	err = suite.OracleKeeper.UpdateOracleInfo(ctx, msgUpdateOracleInfo)
+	suite.Require().NoError(err)
+
+	getOracle, err := suite.OracleKeeper.GetOracle(ctx, suite.oracleAccAddr.String())
+	suite.Require().NoError(err)
+	suite.Require().Equal(suite.newEndpoint, getOracle.Endpoint)
+	suite.Require().Equal(suite.oracleCommissionRate, getOracle.OracleCommissionRate)
+	suite.Require().Equal(ctx.BlockTime(), getOracle.UpdateTime)
+}
+
+func (suite *oracleTestSuite) TestUpdateOracleInfoFailedUpdateTime() {
+	ctx := suite.Ctx
+
+	oracle := types.NewOracle(suite.oracleAccAddr.String(), suite.uniqueID, suite.endpoint, suite.oracleCommissionRate, suite.oracleCommissionMaxRate, suite.oracleCommissionMaxChangeRate, ctx.BlockTime())
+	err := suite.OracleKeeper.SetOracle(ctx, oracle)
+	suite.Require().NoError(err)
+
+	msgUpdateOracleInfo := &types.MsgUpdateOracleInfo{
+		OracleAddress:        suite.oracleAccAddr.String(),
+		Endpoint:             suite.newEndpoint,
+		OracleCommissionRate: suite.newOracleCommissionRate,
+	}
+
+	err = suite.OracleKeeper.UpdateOracleInfo(ctx, msgUpdateOracleInfo)
+	suite.Require().Error(err, types.ErrUpdateOracle)
+	suite.Require().ErrorContains(err, "commission cannot be changed more than once in 24h")
+}
+
+func (suite *oracleTestSuite) TestUpdateOracleInfoFailedGTMaxChangeRate() {
+	ctx := suite.Ctx
+
+	oracle := types.NewOracle(suite.oracleAccAddr.String(), suite.uniqueID, suite.endpoint, suite.oracleCommissionRate, suite.oracleCommissionMaxRate, suite.oracleCommissionMaxChangeRate, ctx.BlockTime())
+	err := suite.OracleKeeper.SetOracle(ctx, oracle)
+	suite.Require().NoError(err)
+
+	msgUpdateOracleInfo := &types.MsgUpdateOracleInfo{
+		OracleAddress:        suite.oracleAccAddr.String(),
+		Endpoint:             suite.newEndpoint,
+		OracleCommissionRate: sdk.NewDecWithPrec(12, 2),
+	}
+	ctx = ctx.WithBlockTime(ctx.BlockTime().Add(30 * time.Hour))
+	err = suite.OracleKeeper.UpdateOracleInfo(ctx, msgUpdateOracleInfo)
+	suite.Require().Error(err, types.ErrUpdateOracle)
+	suite.Require().ErrorContains(err, "commission cannot be changed more than max change rate")
+}
+
+func (suite *oracleTestSuite) TestUpdateOracleInfoFailedGTMaxRate() {
+	ctx := suite.Ctx
+
+	oracle := types.NewOracle(suite.oracleAccAddr.String(), suite.uniqueID, suite.endpoint, suite.oracleCommissionRate, suite.oracleCommissionMaxRate, sdk.NewDecWithPrec(3, 1), ctx.BlockTime())
+	err := suite.OracleKeeper.SetOracle(ctx, oracle)
+	suite.Require().NoError(err)
+
+	msgUpdateOracleInfo := &types.MsgUpdateOracleInfo{
+		OracleAddress:        suite.oracleAccAddr.String(),
+		Endpoint:             suite.newEndpoint,
+		OracleCommissionRate: sdk.NewDecWithPrec(3, 1),
+	}
+	ctx = ctx.WithBlockTime(ctx.BlockTime().Add(30 * time.Hour))
+	err = suite.OracleKeeper.UpdateOracleInfo(ctx, msgUpdateOracleInfo)
+	suite.Require().Error(err, types.ErrUpdateOracle)
+	suite.Require().ErrorContains(err, "commission cannot be more than the max rate")
+}
```

Comment:
Please add test code to down commission.(e.g 10% -> 0%)

Reply:
Since a commission rate of 0% is possible now, I add negative commission rate test code at [7d4ca68](https://github.com/medibloc/panacea-core/pull/540/commits/7d4ca686bba1834fc458c939a864058fa7a0fe73).